lib/ctblmono.gi

```diff
@@ -2176,6 +2176,74 @@ InstallMethod( IsMinimalNonmonomial,
     return false;
 end );
 
+InstallMethod( IsMinimalNonmonomial,
+    "for a non-solvable group",
+    [ IsGroup ],
+    function( K )
+
+    local info, q, p, f;
+
+    if IsSolvableGroup(K) then
+      TryNextMethod();
+    fi;
+
+    # A non-solvable group all of whose proper quotients are monomial
+    # must have every proper quotient solvable. Hence G must be simple.
+    if not IsSimpleGroup(K) then
+      return false;
+    fi;
+
+    # If in addition, every proper subgroup is monomial, then every
+    # proper subgroup must be solvable, so G must be minimal simple.
+    #
+    # A minimal non-simple group must be (by Thompson's N-Groups paper):
+    # * PSL(2,2^f) for f prime, and all such are minimal-non-M groups
+    # * PSL(2,3^f) for f an odd prime, and all such are minimal-non-M groups
+    # * PSL(3,3) which is NOT a minimal-non-M group (contains SL(2,3))
+    # * Sz(2^f) for f an odd prime, and all such are minimal-non-M groups
+    #
+    # The minimal-non-M part follows from the fact that metabelian groups
+    # are monomial groups.
+    # * For PSL(2,q) maximal subgroups are metabelian (since the non-solvable
+    #   maximals are ruled out by minimal simple)
```

Comment:
Maybe expand using the extra information from your excellent PR description? E.g. like this:

Dickson's classification of their subgroups,
all of which are metabelian or non-solvable
```suggestion
    # * For PSL(2,q) Dickson classified their subgroups; the maximal ones are
    #   either metabelian (hence monomial) or non-solvable; but this latter case
    #   is ruled out here by being minimal simple; hence these groups are minimal
    #   non-monomial.
```